### PR TITLE
364 spa support merged

### LINKED ISSUE #655: baseline JSON support for spec-compliance
For spec compliance, this includes supporting JSON responses by Accept header and produces property as well as social integrations

### PR DESCRIPTION
**Pull request by dogeared**
resolves #655

### CODE REVIEW

File: api/src/main/java/com/stormpath/sdk/provider/OauthProvider.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.stormpath.sdk.provider;
+
+/**
+ * @since 1.0.RC8
+ */
+public interface OauthProvider extends Provider {
```
Comment (mraible) [inline api/src/main/java/com/stormpath/sdk/provider/OauthProvider.java]: Should this be named `OAuthProvider`?

File: api/src/main/java/com/stormpath/sdk/provider/RedirectableOauthProvider.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.stormpath.sdk.provider;
+
+/**
+ * @since 1.0.RC8
+ */
+public interface RedirectableOauthProvider extends OauthProvider {
```
Comment (mraible) [inline api/src/main/java/com/stormpath/sdk/provider/RedirectableOauthProvider.java]: Should this be named `RedirectableOAuthProvider`?

Comment (dogeared) [inline api/src/main/java/com/stormpath/sdk/provider/RedirectableOauthProvider.java]: Yes! Please update and merge into the `364-spa-support-merged` branch.

**Comment by mraible** [comment]
FYI... I renamed all `Oauth` classes and interfaces to `OAuth` in https://github.com/stormpath/stormpath-sdk-java/pull/656/commits/985e1f5e3bcfc40d6119ac2c38f8d44629102698

**Comment by mrioan** [comment]
@jarias could you please take a look at this PR? It is so huge that I think that we need several eyes on it. I only found minor things (copyright header and versions in javadoc) which I already fixed. If you find something please commit it yourself and let me know so I can also take a  look at that finding. Thanks!

**Comment by jarias** [comment]
sure thing @mrioan

**Comment by mrioan** [comment]
I found an issue where the cookie for the authenticated account is not saved when using IDSite. Working on it now.

**Comment by mrioan** [comment]
I am still working on this. Please do not merge until I commit the fix. I'll comment here when I do so. Thanks

**Pull request merged**